modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/maven/MTFArtifactUnderTestFinder.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.runtime.module.deployment.impl.internal.maven;
+
+import static java.util.Collections.emptyMap;
+import static java.util.Optional.empty;
+import static java.util.Optional.ofNullable;
+import static org.mule.runtime.deployment.model.api.plugin.ArtifactPluginDescriptor.EXTENSION_BUNDLE_TYPE;
+
+import org.mule.runtime.module.artifact.api.descriptor.BundleDependency;
+import org.mule.runtime.module.artifact.api.descriptor.BundleDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.BundleScope;
+import org.mule.runtime.module.artifact.api.descriptor.ClassLoaderModel;
+
+import java.io.File;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is responsible for finding an artifact and create a {@link BundleDescriptor} to be included in an application
+ * that is going to be used to run MTF test cases.
+ * <p>
+ * It expects the classLoaderModel of the application provides the GAV of the module and an absolute path to were is
+ * located.
+ *
+ * @since 4.2
+ */
+public class MTFArtifactUnderTestFinder {
```

Comment:
We need to take this for another spin, making this more generic (maybe as simple as having an array of 'localDependencies', each providing the uri and gav)

Also, @gsfernandes do you know if adding a local `repository` to the app's pom file would do the trick without having to modify any runtime code? Something like: 

<repositories>
    ......
    <repository>
       <id>my-local-repo</id>
       <url>file://chupito/plugins/chupame-extension/target/munit/repository</url>
    </repository>
    ......
</repositories>

If that works, it will work on any runtime from 4.1.1 and beyond, and is not a big deal since we have complete control over the pom and the target. That repository would have only the plugin under test

Reply:
Sure, that could work too.